### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by yannbf**
## What I did

This PR tries to make the PR template more up to date, including a checklist to help remind contributors of important steps such as writing tests, updating documentation etc.

## How to test

Check the [rich diff](https://github.com/storybookjs/storybook/pull/20617/files?short_path=18813c8#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,20 +1,39 @@
-Issue:
+<!-- Thank you for contributing to Storybook! If your PR is related to an issue, provide the number(s) here, e.g. #1000, #1001 -->
+Issue: #
```
Comment (JReinhold) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: I feel like @shilman always changes this to just be the issue number without the "Issue:", to make it work with the release script. If that is the case, then I guess we should just remove it from here.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
+- [ ] Make sure to add/update documentation regarding your changes
+- [ ] If you are deprecating/removing a feature, make sure to update
+      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)
+
+#### Maintainers
+- [ ] If this PR should be tested against many or all sandboxes,
+      make sure to add the `ci:merged` or `ci:daily` GH label to it.
+- [ ] Make sure this PR contains the correct GH labels
+<!-- with at least one of the following `["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]` -->
```
Comment (JReinhold) [inline .github/PULL_REQUEST_TEMPLATE.md:33]: 1. I still struggle to figure out which of these labels are appropriate for my PRs. I miss a sentence or two about each of them, and I guess this is a pretty good place to have that.?

2. I might be wrong, but I think CI fails if there are multiple of these labels on. If that is true we should rephrase it.

**Review by JReinhold — COMMENTED** [review]
Fantastic!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (vanessayuenn) [inline .github/PULL_REQUEST_TEMPLATE.md:33]:
```suggestion
- [ ] Make sure this PR contains the correct GH labels (**one** of the following: `cleanup`, `feature request`, `bug`, `documentation`, `maintenance`, `dependencies`, `BREAKING CHANGE`,`other`)
```

something like this?

**Review by vanessayuenn — APPROVED** [review]
looks like a good improvement of what we currently have

**Pull request merged**